docs/DeveloperGuide.md

```diff
@@ -1,13 +1,286 @@
-# Developer Guide
+# Developer Guide for MangaTantou
 
-## Acknowledgements
+# Acknowledgements
+MangaTantou's structure has been greatly inspired by the team developers' respective iPs as listed below:
+1. [Donovan](https://github.com/xenthm/ip)
+2. [Sarah](https://github.com/sarahchow03/ip)
+3. [Ian](https://github.com/iaso1774/ip)
+4. [Andy](https://github.com/averageandyyy/ip)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The following third-party libraries were used:
+- [Gson 2.11.0](https://github.com/google/gson): A Java library for converting Java objects to JSON and vice versa, licensed under Apache License 2.0.
 
-## Design & implementation
+Additionally, the following resources/websites were heavily used (they are amazing): 
+- [RegExr](https://regexr.com/): RegExr is an online tool to learn, build, & test Regular Expressions (RegEx / RegExp).
+- [Regex Vis](https://regex-vis.com/): Regex visualizer and editor. 
+- [RegEx101 ](https://regex101.com/): Regex visualizer,editor, and debugger.  
+- [Ashley's PlantUML Doc](https://plantuml-documentation.readthedocs.io/): Documentation about how to use the commands, keywords, options, and other information needed to produce diagrams with PlantUML. 
 
+# Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Overall Architecture
+### Representing Data in MangaTantou
+![AuthorListClass.png](uml/puml/AuthorListClass/AuthorListClass.png)
+The above UML class diagram shows the overall structure of author and manga data an editor using MangaTantou would be interested in. 
+
+> **_NOTE:_** There is circular reference (bidirectional navigability) between `Author` and `Manga` through `MangaList`. 
+
+### Commands 
+![Command Inheritance](/docs/uml/puml/Command/Command.png)
+The current list of viable `Commands` are as follows:
+1. `AddAuthorCommand`
+2. `AddMangaCommand`
+3. `DeleteAuthorCommand`
+4. `DeleteMangaCommand`
+5. `ViewAuthorsCommand`
+6. `ViewMangasCommand`
+7. `AddSalesCommand`
+8. `GreetCommand`
+9. `ByeCommand`
+
+#### Command Structure
+All child `Command` classes must inherit from the abstract `Command` class. Each child class is required to implement the abstract `execute` method.
+
+While child classes may or may not modify the `AuthorList`, they are encouraged to utilize the `Ui` class to interact with users, such as displaying success messages.
+
+#### Guidelines for Future Developers
+When adding new command classes, developers must follow the same method of implementation by 
+inheriting from the abstract `Command` class. Ensure that each new command class includes 
+an implementation of the `execute` method and appropriately interacts with the `Ui` class 
```

Comment:
I think you should not repeat yourself here regarding implementing the "execute" method?

Reply:
Thanks for the comment, I agree that the content is similar in nature regarding the `execute` method but in the Guidelines, I had phrased it and targetted my language to be instructional towards the developers while in Structure, it is more of a generic description. Nonetheless, I hear you and will look into making the text more targetted with reduced repetition. Thanks again!